File: messagematch/src/main/java/org/forwoods/messagematch/generate/nodegenerators/ObjectTypeGenerator.java
package org.forwoods.messagematch.generate.nodegenerators;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.node.JsonNodeFactory;
import com.fasterxml.jackson.databind.node.ObjectNode;

import java.util.LinkedHashMap;
import java.util.Map;

public class ObjectTypeGenerator extends NodeGenerator{
    final Map<String, NodeGenerator> children = new LinkedHashMap<>();
    public ObjectTypeGenerator() {
        super(null);
    }

    @Override
    public JsonNode generate() {
        ObjectNode node = JsonNodeFactory.instance.objectNode();
        for (Map.Entry<String, NodeGenerator> child:children.entrySet()) {
            node.set(child.getKey(), child.getValue().generate());
        }
        return node;
    }

    public void addChild(String name, NodeGenerator node) {
        children.put(name, node);
    }

}

File: messagematch-typescript/src/generate/nodegenerators/ObjectTypeGenerator.ts
import {NodeGenerator} from "./NodeGenerator";
import {ValueProvider} from "../ValueProvider";

export class ObjectTypeGenerator extends NodeGenerator
{
    children:Record<string, NodeGenerator> ={};
    constructor() {
        super(undefined);
    }

    generate(): any {
        const result:any = {};
        for (let childrenKey in this.children) {
            let nodeGen:NodeGenerator = this.children[childrenKey];
            result[childrenKey] = nodeGen.generate();
        }
        return result;
    }

    addChild(name: string, node: NodeGenerator) {
        this.children[name] = node;
    }

}